components/telemetry-operator/internal/fluentbit/config/builder/output.go

```diff
@@ -0,0 +1,115 @@
+package builder
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	telemetryv1alpha1 "github.com/kyma-project/kyma/components/telemetry-operator/apis/telemetry/v1alpha1"
+	"github.com/kyma-project/kyma/components/telemetry-operator/internal/utils/envvar"
+)
+
+func createOutputSection(pipeline *telemetryv1alpha1.LogPipeline, config PipelineConfig) string {
+	output := &pipeline.Spec.Output
+	if len(output.Custom) > 0 {
+		return generateCustomOutput(output, config.FsBufferLimit, pipeline.Name)
+	}
+
+	if output.HTTP.Host.IsDefined() {
+		return generateHTTPOutput(&output.HTTP, config.FsBufferLimit, pipeline.Name)
+	}
+
+	if output.Loki.URL.IsDefined() {
+		return generateLokiOutput(&output.Loki, config.FsBufferLimit, pipeline.Name)
+	}
+
+	return ""
```

Comment:
What if no output is specified? With the current implementation no error is thrown right?

Reply:
True. Re-introduced that.
Problem I found `OutputValidator` and the validation here in the configbuilder are separated and are validating different things.
Same with custom filter validation in the webhook and here. All these validation is distributed and we should have **one** validation logic for both.
Having the same validation in the configbuilder as in the webhook is important because we can not guarantee that the webhook is used.

Reply:
Would like to solve that in a separate PR though.